lib/mastermind.rb

```diff
@@ -0,0 +1,111 @@
+
+class Mastermind
+
+	attr_reader :colors
+	attr_accessor :code
+
+	def initialize 
+		@colors = [ 'r' , 'o', 'y', 'g', 'b', 'p']
+	 	@code = []
+	end
+
+	def print_directions
+		message = """\nWelcome to Mastermind!\n
+		To play: The computer will generate a four digit color code of the following colors
+		Red (r), Orange (o), Yellow (y), Green (g), Blue (b), Purple (p). You will try to
+		guess the code by writing a sequence of four lowercase letters, each representing
+		a color. Colors can be repeated and order does matter in the sequence. The computer
+		will return feedback in the form of pegs: red meaning that a color exists in the
+		secret code and is in the correct location; white meaning that the color is in the 
+		secret code but is in the wrong location. You will have ten guesses.
+
+		Additional commands:
+		/help	: to print the instructions
+		/restart: to generate a new game
+		/quit   : to quit the game\n\n}"""
+
+		puts (message)
+	end
+
+end
+
+
+class SecretCode
+
+	def generate (colors)
+		secret_code = []
+		4.times do |i|
+			secret_code << colors.sample
+		end
+		secret_code
+	end
+
```

Comment:
While there's nothing wrong with generating the secret code this way, I just wanted to point out that `#times` returns an enumerator. That means that the `#map` iterator is available and you could shorten this up to:
      
```rb
def generate(colors)
  4.times.map do |i|
    colors.sample
  end
end
```